app/forms/recovery_code_form.rb

```diff
@@ -25,9 +25,9 @@ def valid_recovery_code?
     recovery_code_generator.verify(code)
   end
 
-  def result
+  def extra_analytics_attributes
     {
-      success: success,
+      method: 'recovery code',
```

Comment:
I think the smell I am sensing here is that it's a "magic number" 

> Unique values with unexplained meaning or multiple occurrences which could (preferably) be replaced with named constants

in this case, the meaning of this string is not 100% clear to me so it could be clarified with a constant. Then, we can also ref that constant in the related spec! What do you think?